tests/test_cache.py

```diff
@@ -0,0 +1,215 @@
+import httpretty
+import json
+import os
+import pickle
+import redis
+import requests
+import shutil
+import struct
+import time
+import unittest
+from cachecontrol.caches.file_cache import FileCache
+from cachecontrol.caches.redis_cache import RedisCache
+from canonicalwebteam.http import CachedSession
+from datetime import timedelta
+from mockredis import mock_redis_client
+from unittest.mock import patch
+
+
+file_cache_directory = ".testcache"
+
+
+class MockRedisSingleton:
+    def __init__(self):
+        self.redis_client = None
+
+    def mock_redis_client_singleton(self, *args, **kwargs):
+        if not self.redis_client:
+            self.redis_client = mock_redis_client()
+        return self.redis_client
+
+
+class TestCachedSession(unittest.TestCase):
+    mock_redis_singleton = MockRedisSingleton()
+
+    def tearDown(self):
+        try:
+            shutil.rmtree(file_cache_directory)
```

Comment:
Since this is only relevant to the `test_file_cache` function, why not do the teardown step in that function itself?

Reply:
This is the default caching mechanism in the CachedSession. So every function in the tests uses the file cache.
I could switch this to an in-memory cache (maybe a configuration parameter to the CachedSession), but I wanted to replicate the default situation that a running app will have.

If we never use the in-memory based approach via a DictCache outside of testing though, maybe it is overloading the module a bit if I add this?